client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -46,39 +42,38 @@ function pasteTrack( editor ) {
 		return SOURCE_UNKNOWN;
 	}
 
-	function flattenArray( orgTypes ) {
-		// As per https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
-		// types should be an array containing the available formats that were set on copy.
-		// Some browsers, as Firefox, return a DOMStringList instead, see
-		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
-		let types = [];
-		if ( orgTypes.contains ) {
-			types = Array.from( orgTypes );
-		} else {
-			types = orgTypes.slice();
-		}
-		return types.reduce( ( a, b ) => {
-			return a + ', ' + b;
-		} );
+	function toString( types ) {
+		// types should be an array. Some browsers, as Firefox, return a DOMStringList instead.
+		// We need coverage for thoses cases, hence the Array.from conversion.
+		//
+		// https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
+		// DOMStringListhttps://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
+		return Array.from( types ).join( ', ' );
 	}
 
-	function onPasteEditor( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'tinymce' );
+	function onPasteFromTinyMCEEditor( event ) {
+		recordEvent( 'tinymce-editor', toString( event.clipboardData.types ) );
 	}
 
-	function onPasteTextArea( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'html' );
+	function onPasteFromHTMLEditor( event ) {
+		recordEvent( 'html-editor', toString( event.clipboardData.types ) );
 	}
 
-	// Bind paste event listeners to both Visual and HTML editors
-	editor.on( 'paste', onPasteEditor );
+	debug( 'bind listeners for paste event' );
```

Comment:
Unless there is a guide in calypso that recommends deleting the `debug` statements, my personal preference is for maintain them as they are, as they are useful not only in the development phase, but in the debugging as well - when the worst happens, better to have the tools already in place.